ecosystem/indexer-grpc/indexer-grpc-table-info/src/table_info_parser.rs

```diff
@@ -0,0 +1,284 @@
+
+                    if retries >= DEFAULT_NUM_RETRIES {
+                        error!(
+                            starting_version = batch.start_version,
+                            num_transactions = batch.num_transactions_to_fetch,
+                            error = format!("{:?}", err),
+                            "Could not fetch transactions: retries exhausted",
+                        );
+                        panic!(
+                            "Could not fetch {} transactions after {} retries, starting at {}: {:?}",
+                            batch.num_transactions_to_fetch, retries, batch.start_version, err
+                        );
+                    } else {
+                        error!(
+                            starting_version = batch.start_version,
+                            num_transactions = batch.num_transactions_to_fetch,
+                            error = format!("{:?}", err),
+                            "Could not fetch transactions: will retry",
+                        );
+                    }
+                    tokio::time::sleep(Duration::from_millis(300)).await;
+                },
+            }
+        }
+    }
+
+    fn parse_table_info(
+        context: Arc<Context>,
+        raw_txns: Vec<TransactionOnChainData>,
+        db_writer: Arc<dyn DbWriter>,
+    ) -> Result<(), Error> {
+        if raw_txns.is_empty() {
+            return Ok(());
+        }
+
+        let start_millis = chrono::Utc::now().naive_utc();
+        let first_version = raw_txns.first().map(|txn| txn.version).unwrap();
+        let ledger_chain_id = context.chain_id().id();
+
+        let write_sets: Vec<WriteSet> = raw_txns.iter().map(|txn| txn.changes.clone()).collect();
+        let write_sets_slice: Vec<&WriteSet> = write_sets.iter().collect();
+        db_writer
+            .clone()
+            .index_table_info(context.db.clone(), first_version, &write_sets_slice)
+            .expect("Failed to process write sets and index to the table info rocksdb");
+        let fetch_millis = (chrono::Utc::now().naive_utc() - start_millis).num_milliseconds();
+
+        info!(
+            table_info_first_version = first_version,
+            ledger_chain_id = ledger_chain_id,
+            write_sets_bcs_size = bcs::serialized_size(&write_sets).unwrap(),
+            table_info_parsing_millis_per_batch = fetch_millis,
+            num_transactions = raw_txns.len(),
+            "Table info parsed successfully"
+        );
+
+        Ok(())
+    }
+
+    pub fn set_highest_known_version(&mut self) -> anyhow::Result<()> {
+        let info = self.context.get_latest_ledger_info_wrapped()?;
+        self.highest_known_version = info.ledger_version.0;
+        Ok(())
+    }
+
+    /// Will keep looping and checking the latest ledger info to see if there are new transactions
+    /// If there are, it will set the highest known version
+    async fn ensure_highest_known_version(&mut self) {
```

Comment:
This whole function is unnecessary and frankly this is needlessly complex. This like it was copied from the original indexer; we should've removed it after we changed the implementation to batch based.

Reply:
this fn is needed to make sure ledger version is not 0, nor is it less than the `current_version`, otherwise, we're processing the old txns or txns that don't even exist.even if its batch based implementation, we still need to wait on till ledger version's ahead of us.